BrightFutures/AsyncType+Debug.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AsyncType+Debug.swift
+//  BrightFutures
+//
+//  Created by Oleksii on 23/09/2016.
+//  Copyright © 2016 Thomas Visser. All rights reserved.
+//
+
+import Result
+
+public protocol LoggerType {
+    func log(message: String)
+}
+
+fileprivate struct Logger: LoggerType {
+    func log(message: String) {
+        print(message)
+    }
+}
+
+public extension AsyncType where Value: ResultProtocol {
```

Comment:
When I read this, I was initially a bit confused by why we'd need the restriction on Value. But the reason is because andThen is defined on `AsyncType where Value: ResultProtocol`. There's no real reason why this is the case. Could you move `andThen` to an unrestricted extension of `AsyncType` and remove the restriction here as well.